src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs

```diff
@@ -38,6 +44,17 @@ public class FirebaseMessagingService : Firebase.Messaging.FirebaseMessagingServ
     public override void OnNewToken(string token)
     {
         Log.LogDebug("OnNewToken: '{Token}'", token);
+        if (TryGetScopedServices(out var services)) {
```

Comment:
ScopedServices are initialized when MauiBlazorApp is initialized, so UI is loaded. For case when we have UI loaded, NotificationsPermissionBanner invokes registering device.
If we want to react on new token message when ui is not loaded we need to use services from root container only.
Also I think we should check if notification permission is granted.

Reply:
We don't have a session in the root container, so do you have any ideas 🤔